pull_request_template.md

```diff
@@ -19,3 +19,25 @@ _How was this tested? Have new unit tests been included?_
 
 ## Notes
 _Optional extra information._
+
+---
+
+## PR review checklist
+
+- Is the PR against an appropriate branch?
+- Does the code adhere to coding guidelines?
+- Does the code do what it claims to do?
+- Is the code readable?
+  Is it appropriately documented?
+- Is the logic in the correct place?
+- Was the code tested appropriately?
+  Are there unit tests?
+  Are unit tests self-contained and non-redundant?
+- Did Travis CI pass for the code in the PR?
+  Is Codecov acceptable based on the included/updated unit tests?
+- Will the new code fail gracefully?
+- Does the code follow good practice for writing performant code
+  (e.g. using a database transaction rather than repeated queries outside of a transaction)?
+- Does it bring in an unnecessary dependency?
+- If you are reviewing a new analysis, is it future-proof and pluggable?
+- Does the PR meet agile guidelines?
```

Comment:
```suggestion
- [ ] Is the PR against an appropriate branch?
- [ ] Does the code adhere to coding guidelines?
- [ ] Does the code do what it claims to do?
- [ ] Is the code readable?
  Is it appropriately documented?
- [ ] Is the logic in the correct place?
- [ ] Was the code tested appropriately?
  Are there unit tests?
  Are unit tests self-contained and non-redundant?
- [ ] Did Travis CI pass for the code in the PR?
  Is Codecov acceptable based on the included/updated unit tests?
- [ ] Will the new code fail gracefully?
- [ ] Does the code follow good practice for writing performant code
  (e.g. using a database transaction rather than repeated queries outside of a transaction)?
- [ ] Does it bring in an unnecessary dependency?
- [ ] If you are reviewing a new analysis, is it future-proof and pluggable?
- [ ] Does the PR meet agile guidelines?
```
Purely for aesthetic reasons, I cannot see if this suggestion works, but this should provide visual "check boxes"

Reply:
Just to add to @CristiGuijarro's comment (which I like and suggest to follow), I think it would be best if all questions related with the same task (or check box) are in the same line (not sure but I suspect that is not the case at the moment).

Reply:
I prefer when they are in separate lines, potentially formatted with indentations to make it more visually appealing and easier to perceive as belonging to the same group of questions 😸
Indentations aside (not sure how they'd look like in this setting), I find it easier to "tick off" if having one item (question) per line than multiple ones.

Reply:
@JAlvarezJarreta, it's true that for checkboxes with multiple questions, I had put these on separate lines. Though in other Markdown flavours these might be merged onto a single line, in GitHub Markdown they are not.

@ipilizota, I've tried to group questions only where they are related elements of one overarching question (e.g. are there unit tests and do they meet our needs?).